lib/interferon.rb

```diff
@@ -149,15 +150,14 @@ def read_groups(sources)
     def read_hosts(sources)
       statsd.gauge('hosts.sources', sources.count)
 
-      hosts = []
       loader = HostSourcesLoader.new([@alerts_repo_path])
-      loader.get_all(sources).each do |source|
+      hosts = Parallel.map(loader.get_all(sources), in_threads: @read_hosts_threads) do |source|
         break if @request_shutdown
```

Comment:
I believe this needs to be changed to `raise Parallel::Kill if @request_shutdown`

Reply:
Yeah, I think you're right. I'll make this change.